sympy/sets/sets.py

```diff
@@ -1883,6 +1883,10 @@ def as_relational(self, symbol):
     def compare(self, other):
         return (hash(self) - hash(other))
 
+    def _eval_evalf_options(self, prec, options):
+        from sympy.core.evalf import evalf_options
+        return FiniteSet(*[evalf_options(elem, prec, options) for elem in self])
+
     def _eval_evalf(self, prec):
         return FiniteSet(*[elem.evalf(n=prec_to_dps(prec)) for elem in self])
```

Comment:
One of these methods should be calling the other.

It probably makes sense to add a default implementation of `_eval_evalf` to `EvalfMixin` that calls `_eval_evalf_options`.

Reply:
We can change implementation of `_eval_evalf` in Sympy codebase but changing default `_eval_evalf` won't help because they are overridden in every class.


We can't even call `_eval_evalf_options`  from `_eval_evalf` because we won't have `options` parameter.

Reply:
It's helpful to add in the default implementation because it means that if downstream code is calling `_eval_evalf` directly on a sympy class then that will still work. It also means that we don't need to duplicate the methods like above.

Reply:
Current changes in `.evalf()` would call `_eval_evalf_options` first , it fails then would try with `_eval_evalf` making both work.

I added back `_eval_evalf` because if some method class isn't updated accordingly it doesn't fail, to maintain backward compatibility .

We can't call `_eval_evalf_options` in any way from `_eval_evalf` because of missing parameter `options` and would give `TypeError`, whole purpose of `_eval_evalf_options` is to handle `options` because `_eval_evalf` can not.

Reply:
The default `_eval_evalf` method in `EvalfMixin` could be made to call `_eval_evalf_options` passing the default values for the options.

Reply:
If we change the default `_eval_evalf` method in `EvalfMixin` still we would need to duplicate the method because every every class calls `evalf` on its args differently in `_eval_evalf` that's why they need to be overridden in every class.

as in `FiniteSet` it is -
https://github.com/sympy/sympy/blob/d053c1d4c3ca5420851dcd9a30dfc0e7cee98a48/sympy/sets/sets.py#L1908

but in it `Interval` -
https://github.com/sympy/sympy/blob/d053c1d4c3ca5420851dcd9a30dfc0e7cee98a48/sympy/sets/sets.py#L1106

Reply:
As long as EvalfMixin makes sure that both of those end up calling `_eval_evalf_options` then there should be no need to duplicate anything.

Reply:
Yes, got it. Sorry mistook `_eval_evalf_options` with `evalf_options`.